client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/internal/JooqBackend.kt

```diff
@@ -0,0 +1,69 @@
+package app.cash.backfila.client.misk.jooq.internal
+
+import app.cash.backfila.client.misk.Description
+import app.cash.backfila.client.misk.jooq.ForBackfila
+import app.cash.backfila.client.misk.jooq.JooqBackfill
+import app.cash.backfila.client.misk.spi.BackfilaParametersOperator
+import app.cash.backfila.client.misk.spi.BackfillBackend
+import app.cash.backfila.client.misk.spi.BackfillRegistration
+import com.google.inject.Inject
+import com.google.inject.Injector
+import com.google.inject.Singleton
+import com.google.inject.TypeLiteral
+import com.squareup.moshi.Types
+import java.lang.reflect.ParameterizedType
+import kotlin.reflect.KClass
+
+@Singleton
+class JooqBackend @Inject constructor(
```

Comment:
@mpawliszyn is there a good reason for each backend to duplicate all this code?

Reply:
Some of it is in order for Backfills to be easy to write if I remember correctly.